src/backend/cuda_backend.cc

```diff
@@ -0,0 +1,49 @@
+#include "backend/cuda_backend.h"
+
+#include <cuda_runtime.h>
+
+#include "ranges.h"
+
+namespace celerity::detail::backend_detail {
+
+void memcpy_strided_device_cuda(sycl::queue& queue, const void* source_base_ptr, void* target_base_ptr, size_t elem_size, const sycl::range<1>& source_range,
+    const sycl::id<1>& source_offset, const sycl::range<1>& target_range, const sycl::id<1>& target_offset, const sycl::range<1>& copy_range) {
+	(void)queue;
+	const size_t line_size = elem_size * copy_range[0];
+	const auto ret = cudaMemcpy(static_cast<char*>(target_base_ptr) + elem_size * get_linear_index(target_range, target_offset),
+	    static_cast<const char*>(source_base_ptr) + elem_size * get_linear_index(source_range, source_offset), line_size, cudaMemcpyDefault);
+	if(ret != cudaSuccess) throw std::runtime_error("cudaMemcpy failed");
+	// Classic CUDA footgun: Memcpy is not always synchronous (e.g. for D2D)
+	cudaStreamSynchronize(0);
```

Comment:
`cudaStreamSynchronize` can also fail (especially if the `cudaMemcpy` is async). 

Suggestion / backport from distr-gather for all CUDA calls in this file:

```c++
// clang-format off
#define CELERITY_STRINGIFY2(f) #f
#define CELERITY_STRINGIFY(f) CELERITY_STRINGIFY2(f)
#define CELERITY_CUDA_CHECK(f, ...)                                                               \
	if (const auto cuda_check_result = (f)(__VA_ARGS__); cuda_check_result != cudaSuccess) {      \
		CELERITY_CRITICAL(CELERITY_STRINGIFY(f) ": {}", cudaGetErrorString(cuda_check_result));   \
		abort();                                                                                  \
	}
// clang-format on

CELERITY_CUDA_CHECK(cudaStreamSynchronize, 0);
```

Reply:
Good call, fixed!